### DISCUSSION TITLE
Help a smooth-brain w/ some style tips

### DISCUSSION THREAD

**Discussion by packetpilot** [EdgeTX/edgetx#4537] [Q&A] [answered]
Hi there. Long time user (on my Taranis, at least), first time contributor (well, in progress).

I've dug a bit for a style guide and have come up short. Maybe there are zero cares & one of the jobs tidies things up to conform to some standard, or maybe there aren't many cares at all, especially when it comes to the shell scripts (which is likely where any of my substantive contributions would initially be).

For context, I've done a lot of bash plumbing in my day & some more elegant things as well, but mostly ops-y toil & custodial things to support embedded SWEs (automation in SCM land & dev endpoints) and aero wizards (think wind tunnel, but only real wind is behind the racks of the cluster nodes), and also more boring internet-scale stuff that never held a candle to working in aerospace.

I'd like to contribute to etx and would like to start off on the right foot. I'm sick of letting my X10S Express go another three years unused now that I'm doing some INAV rover work, and due to a sensory incompatibility (_for me_) w/ EdgeTX on the X10S Express, I've been using ETHOS under duress, and would like to stop doing so. (It's not a _bad_ OS per se, but there are many reasons I'd rather use EdgeTX, and ETHOS handles the LCD backlight "correctly" and, imho, perfectly).

So back to the point: is there some style guide I'm missing? Is there one (maybe [this one](https://google.github.io/styleguide/)) that is adopted, at least loosely?

This is as far as I could get w/r/t contributing to the ETHOS "source" by the way, other than providing feedback of course for source I can't actually alter... _a repo for feedback_? _INACTIVITY FOR LONG TIME?_ sensor lost _sensory overload_?!?! 

## F THIS!
![image](https://github.com/EdgeTX/edgetx/assets/1474054/b18322ee-b68a-4505-a578-22eebcf1239f)

**Comment by pfeerick** [comment, marked as answer]
More the merrier. If you're thinking of getting started with something, join us on in the `#dev` channel on the [EdgeTX Discord server](https://discord.gg/wF9wUKnZ6H). We also have a regular voice/video chat on Discord to catch up with everyone working on the project and see how things are going, and help if get stuck. Otherwise, for code formatting, we generally use the Google Cpp styling... per https://github.com/EdgeTX/edgetx/blob/main/.clang-format

Otherwise, we probably still somewhat follow this loosely : https://github.com/opentx/opentx/wiki/OpenTX-Code-Style-Guide

Due to the age of the code, sometimes it's just a matter of fitting in with the existing structure of the code, or if you're in the mood for it, re-writing  and modernising whole chunks of it to take benefit of new C++ structures and capabilities. 

As far as the shell scripts... from my perspective... as long as they work, are understandable enough, and ShellCheck approves... I'm happy 🤭

**Reply by packetpilot** [reply]
Thanks @pfeerick -- I am somehow unsurprised that I missed the whole code style guide section of the wiki, but at least I _did_ pick up on gstyle for cpp in some job yaml.

> or if you're in the mood for it, re-writing and modernising whole chunks of it to take benefit of new C++ structures and capabilities.

I'm no SWE, just a digital plumber, but perhaps I can toy with gh jobs to at least bring things in line with gstyle. Most of my work was on the gitlab pipelines flavor, but a github job or three came out of me at lastjob.

Please feel free to permaban me if I ever start pitching a move to gitlab <=> github "for the pipeline capabilities alone" 😆

I'll try to choke down discord participation -- I know I know, putting all sorts of banter into "other people's computers" is all the rage these days, and at least discord isn't super greedy on the PII front. I do have to wonder though whose NLP models are benefiting from all the discourse on their platform though. (Hey, at least it's not Salesforce ....yet!)

Once I make more progress on this X10S display situation I'll pop in. Thanks again!

**Reply by pfeerick** [reply]
Nah, you're all good... I've used Gitlab a bit also, as well as GitHub, and keep coming back to github... just because... but both services have their good and bad points. 

If you can add something for styling to the actions, even if it is just a check that proposes formatting changes/throws a fit if code isn't conforming with the formatting, that would be great. No pressure though, I'll just keep forcing formatting changes down peoples throats if they don't comply 🤪
